### PR TITLE
improve(MulticallerClient): Remove extra simulation

### PR DESCRIPTION
**Pull request by nicholaspai**
This PR removes a simulation of the batch sent to `Multicall3`. This was originally included because technically its possible that individually transactions will succeed when batched together, they can revert.

This extra simulation has downsides:
- Extra RPC request to simulate transaction ends up leading to slow down in transactions being mined which opens up to ultimate failures like nonce-reuse and front-running
- More logic

With some upsides:
- Catches case where batch fails while individual txns succeed

Ultimately, the dataworker and relayer functions using this code should never be batching together transactions that would fail in a batch but succeed individually, so I think its safe to remove to mitigate the downsides above. Moreover, if the batch fails, we'll hear about it anyways.

The motivation for this PR was an error log that showed the batch failed simulation, which suggests that the individual txn suceeded, but the batch failed. This error did not persist, meaning that the `fillRelay` reverted after succeeding the individual simulation but was probably front-run by the time of the batch txn.

### CODE REVIEW

**Review by mrice32 — APPROVED** [review]
Agreed!

I think this simulation would make more sense if it fell back to running the txns individually, but since we don't do this, I think it's probably okay to just allow the txn, itself to fail.

**Comment by pxrl** [comment]
@nicholaspai I think the reasoning seems OK, though it'd be interesting to quantify the impact of a second-round simulation.

One thing that puzzles me a bit is the description of a fillRelay transaction that passed initial simulation and failed once bundled via multicall3. In which cases do we try to submit relay fills via multicall3? afaik we should only fill slow relays via multicall3, and in those cases we wouldn't expect to see race conditions or front-running by third parties.

**Review by james-a-morris — APPROVED** [review]
Left a question - otherwise LGTM

**Comment by nicholaspai** [comment]
> @nicholaspai I think the reasoning seems OK, though it'd be interesting to quantify the impact of a second-round simulation.
> 
> One thing that puzzles me a bit is the description of a fillRelay transaction that passed initial simulation and failed once bundled via multicall3. In which cases do we try to submit relay fills via multicall3? afaik we should only fill slow relays via multicall3, and in those cases we wouldn't expect to see race conditions or front-running by third parties.

We will never fill relays via multicall3, that's a mistake. We'll use the `multicall` method on the SpokePool instead

File: src/clients/MultiCallerClient.ts [hunk 1]
```diff
@@ -179,35 +179,14 @@ export class MultiCallerClient {
...
-      const successfulTxns = _valueTxns.concat(_txns);
-      successfulTxns.forEach((txn, idx) => {
+      txnRequests.forEach((txn, idx) => {
         mrkdwn += `  *${idx + 1}. ${txn.message || "No message"}: ${txn.mrkdwn || "No markdown"}\n`;
       });
       this.logger.info({
         at: "MultiCallerClient#executeTxnQueue",
-        message: `${successfulTxns.length}/${nTxns} ${networkName} transaction simulation(s) succeeded!`,
+        message: `${txnRequests.length}/${nTxns} ${networkName} transaction simulation(s) succeeded!`,
         mrkdwn,
```
[the remark this answers is not in the document]
Comment (nicholaspai) [reply src/clients/MultiCallerClient.ts:190]: This is a cool idea, could you propose in a separate pr?

**Pull request merged**